File: Assigment2/MusicStudioRentalApp/app/src/main/java/com/example/musicstudiorental/models/RentalItem.java
package com.example.musicstudiorental.models;

import android.os.Parcel;
import android.os.Parcelable;

public class RentalItem implements Parcelable {
    private String name;
    private float rating;
    private String multiChoiceAttribute;
    private int pricePerMonth;

    public RentalItem(String name, float rating, String multiChoiceAttribute, int pricePerMonth) {
        this.name = name;
        this.rating = rating;
        this.multiChoiceAttribute = multiChoiceAttribute;
        this.pricePerMonth = pricePerMonth;
    }

    protected RentalItem(Parcel in) {
        name = in.readString();
        rating = in.readFloat();
        multiChoiceAttribute = in.readString();
        pricePerMonth = in.readInt();
    }

    public static final Creator<RentalItem> CREATOR = new Creator<RentalItem>() {
        @Override
        public RentalItem createFromParcel(Parcel in) {
            return new RentalItem(in);
        }

        @Override
        public RentalItem[] newArray(int size) {
            return new RentalItem[size];
        }
    };

    public String getName() {
        return name;
    }

    public float getRating() {
        return rating;
    }

    public String getMultiChoiceAttribute() {
        return multiChoiceAttribute;
    }

    public int getPricePerMonth() {
        return pricePerMonth;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeFloat(rating);
        dest.writeString(multiChoiceAttribute);
        dest.writeInt(pricePerMonth);
    }
}

File: Assigment2/app/src/main/java/com/example/musi/models/RentalItem.kt
package com.example.musi.models

import android.os.Parcel
import android.os.Parcelable

data class RentalItem(
    val name: String,
    val rating: Float,
    val multiChoiceAttribute: String,
    val pricePerMonth: Float,
    val imageResource: Int
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString() ?: "",
        parcel.readFloat(),
        parcel.readString() ?: "",
        parcel.readFloat(),
        parcel.readInt()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeFloat(rating)
        parcel.writeString(multiChoiceAttribute)
        parcel.writeFloat(pricePerMonth)
        parcel.writeInt(imageResource)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<RentalItem> {
        override fun createFromParcel(parcel: Parcel): RentalItem {
            return RentalItem(parcel)
        }

        override fun newArray(size: Int): Array<RentalItem?> {
            return arrayOfNulls(size)
        }
    }
}